src/module.ts

```diff
@@ -2402,26 +2386,18 @@ export function readString(ptr: usize): string | null {
       if ((cp & 0xF8) == 0xF0) {
         cp = ((cp & 7) << 18) | (u1 << 12) | (u2 << 6) | u3;
       } else {
-        u4 = binaryen.__i32_load8_u(ptr++) & 63;
-        if ((cp & 0xFC) == 0xF8) {
-          cp = ((cp & 3) << 24) | (u1 << 18) | (u2 << 12) | (u3 << 6) | u4;
-        } else {
-          u5 = binaryen.__i32_load8_u(ptr++) & 63;
-          cp = ((cp & 1) << 30) | (u1 << 24) | (u2 << 18) | (u3 << 12) | (u4 << 6) | u5;
-        }
+        assert("Invalid UTF8 sequence during readString");
       }
     }
-    arr.push(cp);
```

Comment:
The important part for the fix appears to be removing this line, that correct? If so, can we make this PR just remove the line as a "fix", before merging the larger one as "refactor"?

Reply:
It will be great if this PR merged first. Then I just sync PR with refactoring with master